src/ds/logger.h

```diff
@@ -117,17 +132,64 @@ namespace logger
     }
   };
 
+  class formatter
+  {
+  public:
+    static void format(
+      const std::string& file_name,
+      int line_number,
+      const std::string& ll_str,
+      const std::string& msg,
+      const std::string& host_ts,
+      const std::optional<std::string>& enclave_ts = std::nullopt)
+    {
+      nlohmann::json j;
+      j["file"] = file_name;
+      j["number"] = line_number;
+      j["level"] = ll_str;
+      j["msg"] = msg;
+      j["h_ts"] = host_ts;
+      if (enclave_ts.has_value())
+      {
+        j["e_ts"] = enclave_ts.value();
+      }
+
+      dump(j.dump(), config::custom_path());
+    }
+
+    static void dump(const std::string& msg, const std::string& file)
+    {
+      using namespace std;
+      ofstream f(file, ios_base::app);
+      f << msg << endl;
+      if (!f)
+        throw logic_error("Failed to write to file: " + file);
+    }
+  };
+
   class LogLine
   {
+  public:
+    std::string file_name;
+    int line_number;
+    std::string file_line;
+    std::string ll_str;
+    std::string msg;
+
   private:
     friend struct Out;
     std::ostringstream ss;
     Level log_level;
 
   public:
-    LogLine(Level ll, const char* file_name, int line_number) : log_level(ll)
+    LogLine(Level ll, const char* file_name, int line_number) :
+      file_name(file_name),
+      line_number(line_number),
+      log_level(ll)
     {
-      const auto file_line = fmt::format("{}:{}", file_name, line_number);
+      ll_str = config::to_string(ll);
+      const char* fn = file_name;
+      file_line = fmt::format("{}:{}", file_name, line_number);
```

Comment:
Why do we convert the log level to string here? We already have a `log_level` member - if we make that public, then the formatter can convert it to string. Similarly with `file_line` - we already store the line_number and file_name, any further formatting should be left to the formatter.

Reply:
The file line is shortened here and that is going into the "normal" logs so I thought it should stay as is. Making the log level a string here means that we don't need to do it for each logger

Reply:
I agree with Eddy, I think the file line shortening is not something we want in the machine logs in any case, it's only done for alignment reasons.

Reply:
ok but then you need to have the shortening twice, once in each write() function which is duplicating code, or create a static method in LogLine or config that will do the shortening

Reply:
Why do we need the shortening twice?